File: Laborator/CSharp/AgentieTurism/Domain/Ticket.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using AgentieTurism.Models;

namespace AgentieTurism.Models
{
    public class Ticket : IIdentifiable<int>
    {
        public int Id { get; set; }
        public Flight Flight { get; set; }
        public string ClientName { get; set; }
        public string TuristsName { get; set; }
        public string ClientAddress { get; set; }
        public int SeatsNumber { get; set; }

        public Ticket(Flight flight, string clientName, string turistsName, string clientAddress, int seatsNumber)
        {
            Flight = flight;
            ClientName = clientName;
            TuristsName = turistsName;
            ClientAddress = clientAddress;
            SeatsNumber = seatsNumber;
        }

        public override string ToString()
        {
            return $"Ticket {{ Id={Id}, Flight={Flight}, ClientName={ClientName}, TuristsName={TuristsName}, ClientAddress={ClientAddress}, SeatsNumber={SeatsNumber} }}";
        }
    }
}

File: Laborator/Java/AgentieTurism/src/main/java/ro/mpp2024/domain/Ticket.java
package ro.mpp2024.domain;

public class Ticket implements Identifiable<Integer>{
    private int id;
//    private int idFlight;

    private Flight flight;
    private String clientName;
    private String turistsName;
    private String clientAddress;
    private int seatsNumber;

    public Ticket(Flight flight, String clientName, String turistsName, String clientAddress, int seatsNumber) {
        this.flight = flight;
        this.clientName = clientName;
        this.turistsName = turistsName;
        this.clientAddress = clientAddress;
        this.seatsNumber = seatsNumber;
    }

    @Override
    public void setId(Integer id) {
        this.id = id;
    }

    @Override
    public Integer getId() {
        return id;
    }

    public Flight getFlight() {
        return flight;
    }

    public void setFlight(Flight flight) {
        this.flight = flight;
    }

    public String getClientName() {
        return clientName;
    }

    public void setClientName(String clientName) {
        this.clientName = clientName;
    }

    public String getTuristsName() {
        return turistsName;
    }

    public void setTuristsName(String turistsName) {
        this.turistsName = turistsName;
    }

    public String getClientAddress() {
        return clientAddress;
    }

    public void setClientAddress(String clientAddress) {
        this.clientAddress = clientAddress;
    }

    public int getSeatsNumber() {
        return seatsNumber;
    }

    public void setSeatsNumber(int seatsNumber) {
        this.seatsNumber = seatsNumber;
    }

    @Override
    public String toString() {
        return "Ticket{" +
                "id=" + id +
                ", flight=" + flight +
                ", clientName='" + clientName + '\'' +
                ", turistsName='" + turistsName + '\'' +
                ", clientAddress='" + clientAddress + '\'' +
                ", seatsNumber=" + seatsNumber +
                '}';
    }
}
